File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/QuestionnaireResponseEditorDetailsDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import net.dnadas.training_portal.dto.user.UserResponsePublicDto;
import net.dnadas.training_portal.model.group.project.questionnaire.QuestionnaireStatus;

import java.util.List;

public record QuestionnaireResponseEditorDetailsDto(
  Long id, String name, String description,
  QuestionnaireStatus status,
  Integer maxPoints,
  UserResponsePublicDto createdBy, String createdAt,
  UserResponsePublicDto updatedBy, String updatedAt,
  List<QuestionResponseEditorDto> questions) {
}

File: frontend/src/questionnaires/dto/QuestionnaireResponseEditorDetailsDto.ts
import {QuestionResponseEditorDto} from "./QuestionResponseEditorDto.ts";
import {QuestionnaireStatus} from "./QuestionnaireStatus.ts";
import {UserResponsePublicDto} from "../../user/dto/UserResponsePublicDto.ts";

export interface QuestionnaireResponseEditorDetailsDto {
  id: number;
  name: string;
  description: string;
  status: QuestionnaireStatus;
  maxPoints: number;
  createdBy: UserResponsePublicDto;
  createdAt: Date;
  updatedBy: UserResponsePublicDto;
  updatedAt: Date;
  questions: QuestionResponseEditorDto[];
}
